src/ts/state/reducer.ts

```diff
@@ -840,3 +848,28 @@ const handleSetFilterState = <TPluginState extends IPluginState>(
   ...state,
   filterState
 });
+
+const handleSetTyperighterEnabled = (getIgnoredRanges: TGetSkippedRanges = doNotSkipRanges) => <TPluginState extends IPluginState>(
+  _: Transaction,
+  state: TPluginState,
+  { payload: { typerighterEnabled } }: ActionSetTyperighterEnabled,
+): TPluginState => {
+
+  const matchRequestAction = { type: REQUEST_FOR_DOCUMENT, payload: { requestId: v4(), categoryIds: []}}
+  const updatedState = typerighterEnabled ? {
+    // Run a check if typerighter has been enabled
```

Comment:
Non-blocking – we should not need to spread our state into a new object, as you will always get a new identity from an action handler.